File: 백준/Bronze/15874. Caesar Cipher/Caesar Cipher.c
#include <stdio.h>
#include <ctype.h>

int main() {
    int k, s;
    scanf("%d %d", &k, &s);
    char S[100001];
    scanf(" %[^\n]", S);
    char dap[100001];
    int i;
    for (i = 0; i < s && S[i] != '\0'; i++) {
        if (isalpha(S[i])) {
            if (isupper(S[i])) {
                dap[i] = (char)(((S[i] - 'A' + k) % 26) + 'A');
            } else {
                dap[i] = (char)(((S[i] - 'a' + k) % 26) + 'a');
            }
        } else {
            dap[i] = S[i];
        }
    }
    dap[i] = '\0';  
    printf("%s\n", dap);
    return 0;
}

File: 백준/Bronze/15874. Caesar Cipher/Caesar Cipher.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    k, s = map(int, input().split())
    S = input()
    dap = ''
    for i in range(s):
        if S[i].isalpha():
            if S[i].isupper():
                dap += chr((ord(S[i])-65+k)%26+65)
            else:
                dap += chr((ord(S[i])-97+k)%26+97)
        else:
            dap += S[i]
    print(dap)


if __name__ == '__main__':
    main()